provisioning_templates/snippet/create_users.erb

```diff
@@ -0,0 +1,25 @@
+<%#
+name: create_users
+kind: snippet
+snippet: true
+-%>
+<%- users = @host.owner_type == 'Usergroup' ? @host.owner.all_users : [@host.owner] -%>
+<%- users.each do |user| -%>
+<%=   "useradd #{user.login}" %>
+<%-   if user.ssh_authorized_keys && user.ssh_authorized_keys.any? -%>
```

Comment:
`user.ssh_authorized_keys` will always evaluate to true since it's either empty array or array of keys. If that check was meant to verify the user object responds to that method, `respond_to?` should be used but since templates are versioned I don't think that's needed